include/cuco/detail/static_map/static_map_ref.inl

```diff
@@ -248,7 +250,7 @@ class operator_impl<
     static_assert(cg_size == 1, "Non-CG operation is incompatible with the current probing scheme");
 
     ref_type& ref_       = static_cast<ref_type&>(*this);
-    auto const key       = value.first;
+    auto const key       = thrust::get<0>(thrust::raw_reference_cast(value));
```

Comment:
can you please remind me why `thrust::get` is preferred?

Reply:
This is a leftover from the heterogeneous insert PR. The input type can now be any _tuple-like_ type that is convertible to the map's `value_type`. The `.first` and `.second` members are only defined for _pair-like_ types, so I had to use `get` here instead.

Reply:
I went with the following convention: Use `.first/.second` where possible to indicate that the expected type is a `cuco::pair` - just for sanity checking, and only use `get` when needed aka we're dealing with a user-defined tuple-like type.